components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java

```diff
@@ -71,4 +85,44 @@ public Response systemScopesGet(MessageContext messageContext) throws APIManagem
         }
         return null;
     }
+
+    public Response updateRolesForScope(ScopeListDTO body, MessageContext messageContext)
```

Comment:
This is too much business logic into the REST API layer. Shall we think of splitting this into core methods and calling them from here and reduce the number of lines in here?

Reply:
Done with a7a380e